src/server/pps/server/api_server.go

```diff
@@ -3089,7 +3090,46 @@ func RepoNameToEnvString(repoName string) string {
 	return strings.ToUpper(repoName)
 }
 
-func (a *apiServer) rcPods(ctx context.Context, rcName string) ([]v1.Pod, error) {
+func (a *apiServer) pachdPods(ctx context.Context) ([]v1.Pod, error) {
+	podList, err := a.env.KubeClient.CoreV1().Pods(a.namespace).List(ctx, metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ListOptions",
+			APIVersion: "v1",
+		},
+		LabelSelector: metav1.FormatLabelSelector(metav1.SetAsLabelSelector(map[string]string{"app": "pachd"})),
+	})
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	return podList.Items, nil
+}
+
+func (a *apiServer) rcPods(ctx context.Context, pipelineName string, pipelineVersion uint64) ([]v1.Pod, error) {
+	podList, err := a.env.KubeClient.CoreV1().Pods(a.namespace).List(ctx, metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ListOptions",
+			APIVersion: "v1",
+		},
+		LabelSelector: metav1.FormatLabelSelector(metav1.SetAsLabelSelector(map[string]string{
+			"app":             "pipeline",
+			"pipelineName":    pipelineName,
+			"pipelineVersion": fmt.Sprint(pipelineVersion),
+		})),
+	})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
hmm, have you seen this? I'm confused when List would give not found

Reply:
It was a belt-and-suspenders approach, but no doubt you’re correct in thinking it unneeded. Removed.